src/software/logger/coloured_cout_sink.cpp

```diff
@@ -61,15 +62,41 @@ void ColouredCoutSink::displayColouredLog(g3::LogMessageMover log_entry)
         return;
     }
 
+    std::chrono::_V2::system_clock::time_point current_time =
+        std::chrono::system_clock::now();
+    bool past_time = current_time - LOG_INTERVAL_TIMESTAMP < last_msg_timestamp;
```

Comment:
The variable name `past_time` sounds like it could be a time stamp. Maybe calling it `time_has_elapsed` or something similar would be more clear.